web/js/MapUI/Components/UpdateDate/UpdateDate.js

```diff
@@ -0,0 +1,184 @@
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {
+  findIndex as lodashFindIndex,
+  get as lodashGet,
+} from 'lodash';
+import {
+  getActiveLayers,
+  getAllActiveLayers,
+  getActiveLayerGroup,
+  getGranuleCount,
+} from '../../../modules/layers/selectors';
+import { setStyleFunction } from '../../../modules/vector-styles/selectors';
+import { getSelectedDate } from '../../../modules/date/selectors';
+import * as dateConstants from '../../../modules/date/constants';
+import * as layerConstants from '../../../modules/layers/constants';
+
+const UpdateDate = (props) => {
+  const {
+    action,
+    activeLayers,
+    activeString,
+    compareMapUi,
+    config,
+    dateCompareState,
+    getGranuleOptions,
+    granuleState,
+    isCompareActive,
+    layerState,
+    preloadNextTiles,
+    state,
+    ui,
+    updateLayerVisibilities,
+    vectorStyleState,
+  } = props;
+
+  useEffect(() => {
+    actionSwitch();
+  }, [action]);
+
+  const actionSwitch = () => {
+    if (action.type === dateConstants.SELECT_DATE) {
+      if (ui.processingPromise) {
+        return new Promise((resolve) => {
+          resolve(ui.processingPromise);
+        }).then(() => {
+          ui.processingPromise = null;
+          return updateDate(action.outOfStep);
+        });
+      }
+      return updateDate(action.outOfStep);
+    } if (action.type === layerConstants.TOGGLE_LAYER_VISIBILITY || action.type === layerConstants.TOGGLE_OVERLAY_GROUP_VISIBILITY) {
+      return updateDate();
+    }
+  };
+
+  function findLayerIndex({ id }) {
+    const layerGroup = getActiveLayerGroup(layerState);
+    const layers = layerGroup.getLayers().getArray();
+    return lodashFindIndex(layers, {
+      wv: { id },
+    });
+  }
+
+  function updateVectorStyles (def) {
+    const { vectorStyles } = config;
+    const layerName = def.layer || def.id;
+    let vectorStyleId;
+
+    vectorStyleId = def.vectorStyle.id;
+    if (activeLayers) {
+      activeLayers.forEach((layer) => {
+        if (layer.id === layerName && layer.custom) {
+          vectorStyleId = layer.custom;
+        }
+      });
+    }
+    setStyleFunction(def, vectorStyleId, vectorStyles, null, vectorStyleState);
+  }
+
+  async function updateCompareLayer (def, index, layerCollection) {
+    const { createLayer } = ui;
+    const options = {
+      group: activeString,
+      date: getSelectedDate(dateCompareState),
+      ...getGranuleOptions(granuleState, def, activeString),
+    };
+    const updatedLayer = await createLayer(def, options);
+    layerCollection.setAt(index, updatedLayer);
+    compareMapUi.update(activeString);
+  }
+
+  async function updateDate(outOfStepChange) {
+    const { createLayer } = ui;
+
+    const layerGroup = getActiveLayerGroup(layerState);
+    const mapLayerCollection = layerGroup.getLayers();
+    const layers = mapLayerCollection.getArray();
+    const activeLayers = getAllActiveLayers(state);
```

Comment:
"activeLayers" is aready declared on line 22.

Reply:
When returning variables in `mapStateToProps` you need to list them in the `props` object of the component in order to use them within the component. So in this case I am declaring `activeLayers` as a new variable on line 146 of `mapStateToProps` and using the `getActiveLayers` selector and passing `state` as an argument. Since I want to use this variable  I return it in `mapStateToProps` on line 153 and then I have to list it in `props` on line 22 to use it within the component.